custom_components/tuya_local/devices/beok_tol47_thermostat.yaml

```diff
@@ -0,0 +1,192 @@
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: cold
+          value: Manual mode
+        - dps_val: hot
+          value: Program mode
+        - dps_val: wind
+          value: Temporary program mode
+    - id: 5
+      type: string
+      name: hvac_mode
+      mapping:
+        - dps_val: "0"
+          constraint: power
+          conditions:
+            - dps_val: true
+              value: auto
+            - dps_val: false
+              value_redirect: power
+              value: "off"
+        - dps_val: "1"
+          constraint: power
+          conditions:
+            - dps_val: true
+              value: heat
+            - dps_val: false
+              value_redirect: power
+secondary_entities:
+  - entity: lock
+    name: Child lock
+    category: config
+    dps:
+      - id: 8
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+  - entity: switch 
+    name: Sound
+    icon: "mdi:music-note"
+    category: config
+    dps:
+      - id: 13
+        type: boolean
+        name: switch
+  - entity: number
+    name: Calibration offset
+    category: config
+    icon: "mdi:thermometer"
+    dps:
+      - id: 20
+        type: integer
+        name: value
+        unit: C
+        range:
+          min: -90
+          max: 90
+        mapping:
+          - scale: 10
+  - entity: number
+    name: High temperature limit
+    category: config
+    icon: "mdi:thermometer"
+    dps:
+      - id: 21
+        type: integer
+        name: value
+        unit: C
+        range:
+          min: 15
+          max: 95
+  - entity: select
+    name: Sensor selection
+    category: config
+    dps:
+      - id: 25
+        type: integer
+        name: option
+        mapping:
+          - dps_val: 1
+            icon: "mdi:gauge"
+            value: Internal
+          - dps_val: 2
+            icon: "mdi:thermometer-chevron-down"
+            value: External
+          - dps_val: 3
+            icon: "mdi:thermometer"
+            value: Both
+  - entity: switch
+    name: Anti-frost
+    icon: "mdi:snowflake-melt"
+    category: config
+    dps:
+      - id: 26
+        type: boolean
+        name: switch
+  - entity: select
+    name: Backlight brightness
+    category: config
+    dps:
+      - id: 41
+        type: integer
+        name: option
+        mapping:
+          - dps_val: 0
+            icon: "mdi:lightbulb-on-10"
+            value: None
+          - dps_val: 1
+            icon: "mdi:lightbulb-on-40"
+            value: Low
+          - dps_val: 2
+            icon: "mdi:lightbulb-on-70"
+            value: Middle
+          - dps_val: 3
+            icon: "mdi:lightbulb-on"
+            value: High
+  - entity: select
+    name: Schedule
+    category: config
+    icon: "mdi:calendar-clock"
+    dps:
+      - id: 42
+        type: integer

```

Comment:
Normally enum types in tuya are strings, even if they use numbers as the value (they will have quotes around them in the raw Tuya protocol).  This integration is lenient with string/integer distinction, so it probably works either way but better for the config to use the correct one.

Reply:
Thanks for your comment (very beginner with this), modified variable types (+ add dp16 Fault alarm settings)